src/esp/sensor/CameraSensor.cpp

```diff
@@ -14,62 +14,76 @@
 namespace esp {
 namespace sensor {
 
+void CameraSensorSpec::sanityCheck() {
+  CORRADE_ASSERT(sensorSubType == SensorSubType::Pinhole ||
```

Comment:
When this sanityCheck is called, it will only check if the variables exclusive to this class are legal. The variables in the parent class is thus ignored.

You will have to call the sanityCheck in the base class first. Here, you should call `VisualSensorSpec::sanityCheck()`.

You may do the same thing in the `VisualSensorSpec`.